hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/KafkaOffsetGen.java

```diff
@@ -156,7 +163,13 @@ public static OffsetRange[] computeOffsetRanges(Map<TopicPartition, Long> fromOf
             continue;
           }
 
-          long toOffset = Math.min(range.untilOffset(), range.fromOffset() + eventsPerPartition);
+          long toOffset = -1L;
+          if (range.fromOffset() + eventsPerPartition > range.fromOffset()) {
```

Comment:
NTR(note to reviewer): 
if range.fromOffset() + eventsPerPartition exceeds Long.MAX_VALUE, toOffset will be set to some negative value.